chain/watcher.go

```diff
@@ -77,7 +74,7 @@ func (c *Watcher) index(ctx context.Context, he *HeadEvent) error {
 
 		// If we have a zero confidence window then we need to notify every tipset we see
 		if c.confidence == 0 {
-			if err := c.obs.TipSet(ctx, he.TipSet); err != nil {
+			if err := c.maybeIndexTipSet(ctx, he.TipSet); err != nil {
```

Comment:
We call `maybeIndexTipSet` here to ensure tipsets are processed asynchronously while watching with no confidence. Previously this call would block which is undesired while watching the chain.